optimade/client/client.py

```diff
@@ -148,18 +138,18 @@ def __init__(
         self,
         base_urls: Optional[Union[str, Iterable[str]]] = None,
         max_results_per_provider: int = 1000,
-        headers: Optional[Dict] = None,
+        headers: Optional[dict] = None,
         http_timeout: Optional[Union[httpx.Timeout, float]] = None,
         max_attempts: int = 5,
         use_async: bool = True,
         silent: bool = False,
-        exclude_providers: Optional[List[str]] = None,
-        include_providers: Optional[List[str]] = None,
-        exclude_databases: Optional[List[str]] = None,
+        exclude_providers: Optional[list[str]] = None,
+        include_providers: Optional[list[str]] = None,
+        exclude_databases: Optional[list[str]] = None,
         http_client: Optional[
-            Union[Type[httpx.AsyncClient], Type[requests.Session]]
+            Union[type[httpx.AsyncClient], type[requests.Session]]
         ] = None,
-        callbacks: Optional[List[Callable[[str, Dict], Union[None, Dict]]]] = None,
+        callbacks: Optional[list[Callable[[str, dict], Union[None, dict]]]] = None,
```

Comment:
```suggestion
        callbacks: Optional[list[Callable[[str, dict], Optional[dict]]]] = None,
```

Reply:
I think this was intentional really, I don't like `Optional` when referring to returns (especially here when the majority of callbacks should return `None`); this also matches more closely with the `dict | None` syntax that we will inevitably have to switch to fairly soon